### PR TITLE
Fixed some type mistakes

### PR DESCRIPTION
**Pull request by dw-kihara**
Dialyzer found some type mistakes.

- proplist:proplist() -> proplists:proplist()
- map:map() -> map()
- avoid an undefined record element

### CODE REVIEW

File: src/yamerl_node_map.erl [hunk 1]
```diff
@@ -141,16 +141,15 @@ node_as_proplist_or_map(#yamerl_constr{ext_options = Options}) ->
 
 new_builder(Constr) ->
     Format = node_as_proplist_or_map(Constr),
-    Map = #map_builder{
```
Comment (dw-kihara) [inline src/yamerl_node_map.erl:144]: `data` is defined as `proplists:proplist() | map()`, so making temporary `Map` leaving `data` undefined was illegal.

**Comment by dumbbell** [comment]
Hi! Goot catch, thanks!

Like with #57, could you please add the review comment to the commit message?

**Comment by dumbbell** [comment]
I stopped the test against Erlang 23. You can rebase your branch on top of the latest master and CI should be happy.

**Comment by dw-kihara** [comment]
> Like with https://github.com/yakaz/yamerl/pull/57, could you please add the review comment to the commit message?

Added the summary for the change.

**Comment by dumbbell** [comment]
I merged your pull request, thank you again!

**Pull request merged**